pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -602,7 +602,14 @@ func (c *Controller) updateShootStatusOperationStart(ctx context.Context, g gard
 		})
 }
 
-func (c *Controller) updateShootStatusOperationSuccess(ctx context.Context, g gardencore.Interface, shoot *gardencorev1beta1.Shoot, shootSeedNamespace string, seedName *string, operationType gardencorev1beta1.LastOperationType) (*gardencorev1beta1.Shoot, error) {
+func (c *Controller) updateShootStatusOperationSuccess(
+	ctx context.Context,
+	g gardencore.Interface,
+	op *operation.Operation,
+	shoot *gardencorev1beta1.Shoot,
+	shootSeedNamespace string,
+	seedName *string,
+	operationType gardencorev1beta1.LastOperationType) (*gardencorev1beta1.Shoot, error) {
```

Comment:
For readability:

```suggestion
func (c *Controller) updateShootStatusOperationSuccess(
	ctx context.Context,
	g gardencore.Interface,
	op *operation.Operation,
	shoot *gardencorev1beta1.Shoot,
	shootSeedNamespace string,
	seedName *string,
	operationType gardencorev1beta1.LastOperationType,
) (
	*gardencorev1beta1.Shoot, 
	error,
) {
```

However, you can now probably remove the 

```
	shoot *gardencorev1beta1.Shoot,
	shootSeedNamespace string,
```

arguments as you pass the `*Operation`, so you can access them via `op.Shoot` or `op.Shoot.SeedNamespace`

Also, `	g gardencore.Interface,` is probably not needed as well now that you can access it via `op.K8sGardenClient.GardenCore()`?

Reply:
In some cases the operation would be `nil`